### DISCUSSION TITLE
Generate SQL migration from Entity defenitions.

### DISCUSSION THREAD

**Discussion by marknefedov** [SeaQL/sea-orm#325] [Q&A] [answered]
Is there any plans to provide schema generation from entities like in Hibernate / GORM?

**Comment by billy1624** [comment, marked as answer]
Hi @markusgod, welcome! Yes, you can use [`Schema::create_table_from_entity`](https://docs.rs/sea-orm/0.3.2/sea_orm/schema/struct.Schema.html#method.create_table_from_entity).
https://www.sea-ql.org/SeaORM/docs/write-test/sqlite#setup-database-schema

**Comment by marknefedov** [comment]
Thanks! Will it create also referenced entities and foreign key constraints? I'm also can't find in the docs how to specify default value (I found in the TableCreateStatement example, but it is not clear how to specify all of this from enity).

Here example of what ideally would be nice to be able to generate.
```sql
CREATE TABLE "users" (
  "id" uuid PRIMARY KEY DEFAULT (gen_random_uuid()),
  "name" text NOT NULL,
  "password" text NOT NULL,
  "email" text NOT NULL
);

CREATE TABLE "projects" (
  "id" uuid PRIMARY KEY,
  "name" text NOT NULL,
  "description" text NOT NULL,
  "owner" uuid NOT NULL
);
ALTER TABLE "projects" ADD FOREIGN KEY ("owner") REFERENCES "users" ("id");
```

**Reply by billy1624** [reply]
> Will it create also referenced entities and foreign key constraints?

It will create foreign key constraints if you have Relation defined in your Entity.

Referenced entities are just tables, you have do pass your Entity one by one to [`Schema::create_table_from_entity`](https://docs.rs/sea-orm/0.3.2/sea_orm/schema/struct.Schema.html#method.create_table_from_entity) in order to create all the tables.

> I'm also can't find in the docs how to specify default value.

Default value is currently not supported by [`Schema::create_table_from_entity`](https://docs.rs/sea-orm/0.3.2/sea_orm/schema/struct.Schema.html#method.create_table_from_entity).

**Reply by billy1624** [reply]
You can look at [here](https://github.com/SeaQL/sea-orm/blob/master/tests/common/bakery_chain/schema.rs) as an example.

**Reply by billy1624** [reply]
If you do need the `DEFAULT` value, I suggest you to build the SQL statement manually with [`sea_query::table::TableCreateStatement`](https://docs.rs/sea-query/0.18.2/sea_query/table/struct.TableCreateStatement.html) like below.
https://github.com/SeaQL/sea-orm/blob/b2985a97c803747ce21c5611b7b492a9b004dfd8/tests/common/bakery_chain/schema.rs#L185-L219
